README.md

```diff
@@ -1059,93 +1184,99 @@ _(For more information on these values, checkout Microsoft's documentation [here
-
-    /// A concrete `Input` type for subprocesses that reads input
-    /// from a given type conforming to `StringProtocol`.
-    /// Developers can specify the string encoding to use when
-    /// encoding the string to data, which defaults to UTF-8.
-    public struct StringInput<
-        InputString: StringProtocol & Sendable
-    >: InputProtocol { }
-
-    /// A concrete `Input` type for subprocesses that reads input
-    /// from a given sequence of `UInt8` (such as `Data` or `[UInt8]`).
-    public struct UInt8SequenceInput<
-        InputSequence: Sequence & Sendable
-    >: InputProtocol where InputSequence.Element == UInt8 { }
-
-    /// A concrete `Input` type for subprocesses that accepts input
-    /// from a specified sequence of `Data`. This type should be preferred
-    /// over `Subprocess.UInt8SequenceInput` when dealing with
-    /// large amount input data.
-    public struct DataSequenceInput<
-        InputSequence: Sequence & Sendable
-    >: InputProtocol where InputSequence.Element == Data { }
-
-    /// A concrete `Input` type for subprocesses that reads input
-    /// from a given async sequence of `Data`.
-    public struct DataAsyncSequenceInput<
-        InputSequence: AsyncSequence & Sendable
-    >: InputProtocol where InputSequence.Element == Data { }
-
-    /// A concrete `Input` type for subprocess that indicates that
-    /// the Subprocess should read its input from `StandardInputWriter`.
-    public struct CustomWriteInput: InputProtocol { }
+@available(macOS 9999, *)
+public protocol PipeBasedInputProtocol: InputProtocol {
+    /// The underlying pipe used by this input in order to
+    /// write input to child process
+    var pipe: Mutex<(readEnd: FileDescriptor?, writeEnd: FileDescriptor?)?> { get }
 }
 
+/// A concrete `Input` type for subprocesses that indicates
+/// the absence of input to the subprocess. On Unix-like systems,
+/// `NoInput` redirects the standard input of the subprocess
+/// to `/dev/null`, while on Windows, it does not bind any
+/// file handle to the subprocess standard input handle.
+public struct NoInput: InputProtocol { }
+
+/// A concrete `Input` type for subprocesses that
+/// reads input from a specified `FileDescriptor`.
+/// Developers have the option to instruct the `Subprocess` to
+/// automatically close the provided `FileDescriptor`
+/// after the subprocess is spawned.
+public struct FileDescriptorInput: InputProtocol { }
+
+/// A concrete `Input` type for subprocesses that reads input
+/// from a given type conforming to `StringProtocol`.
+/// Developers can specify the string encoding to use when
+/// encoding the string to data, which defaults to UTF-8.
+public struct StringInput<
+    InputString: StringProtocol & Sendable
+>: PipeBasedInputProtocol { }
+
+/// A concrete `Input` type for subprocesses that reads input
+/// from a given sequence of `UInt8` (such as `Data` or `[UInt8]`).
+public struct UInt8SequenceInput<
```

Comment:
I have the same question here as above - do we really want to provide this? Because accidentally falling back to a byte-by-byte write for each element of some collection could end up being very expensive, and surprising.